src/System.Text.Json/tests/Serialization/EnumTests.cs

```diff
@@ -394,5 +394,146 @@ public static void EnumAsSByteMax()
+            Assert.Equal(long.MaxValue.ToString(), json);
+        }
+
+        [Fact]
+        public static void ToString_MinValue_ReturnsMinValue()
+        {
+            string json = JsonSerializer.ToString((SampleByteEnum)byte.MinValue);
+            Assert.Equal(byte.MinValue.ToString(), json);
+            json = JsonSerializer.ToString((SampleSByteEnum)sbyte.MinValue);
+            Assert.Equal(sbyte.MinValue.ToString(), json);
+            json = JsonSerializer.ToString((SampleEnumUInt16)ushort.MinValue);
+            Assert.Equal(ushort.MinValue.ToString(), json);
+            json = JsonSerializer.ToString((SampleUInt32Enum)UInt32.MinValue);
+            Assert.Equal(UInt32.MinValue.ToString(), json);
+            json = JsonSerializer.ToString((SampleUInt64Enum)UInt64.MinValue);
+            Assert.Equal(UInt64.MinValue.ToString(), json);
+            json = JsonSerializer.ToString((SampleInt64Enum)long.MinValue);
+            Assert.Equal(long.MinValue.ToString(), json);
+        }
+
+        [Theory]
+        [InlineData(byte.MaxValue + 1, (SampleByteEnum)0)]
+        [InlineData(byte.MaxValue + 2, (SampleByteEnum)1)]
+        [InlineData(byte.MaxValue + 13, (SampleByteEnum)12)]
+        [InlineData(byte.MaxValue * 2, (SampleByteEnum)byte.MaxValue - 1)]
+        public static void ToString_Byte(ulong i, SampleByteEnum enumValue)
+        {
+            string json = JsonSerializer.ToString(enumValue);
+            Assert.Equal(((i - 1) % byte.MaxValue).ToString(), json);
+        }
+
+        [InlineData(sbyte.MaxValue + 1, (SampleSByteEnum)0)]
+        [InlineData(sbyte.MaxValue + 2, (SampleSByteEnum)1)]
+        [InlineData(sbyte.MaxValue + 13, (SampleSByteEnum)12)]
+        [InlineData(sbyte.MaxValue * 2, (SampleSByteEnum)sbyte.MaxValue - 1)]
+        public static void ToString_SByte(sbyte i, SampleSByteEnum enumValue)
+        {
+            string json = JsonSerializer.ToString(enumValue);
+            Assert.Equal(((i - 1) % sbyte.MaxValue).ToString(), json);
+        }
+
+        [InlineData((ulong)UInt16.MaxValue + 1, (SampleEnumUInt16)0)]
+        [InlineData((ulong)UInt16.MaxValue + 2, (SampleEnumUInt16)1)]
+        [InlineData((ulong)UInt16.MaxValue + 13, (SampleEnumUInt16)12)]
+        [InlineData((ulong)UInt16.MaxValue * 2, (SampleEnumUInt16)UInt16.MaxValue - 1)]
+        public static void ToString_UInt16(ulong i, SampleUInt32Enum enumValue)
+        {
+            string json = JsonSerializer.ToString(enumValue);
+            Assert.Equal(((i - 1) % UInt16.MaxValue).ToString(), json);
+        }
+
+        [InlineData((ulong)UInt32.MaxValue + 1, (SampleUInt32Enum)0)]
+        [InlineData((ulong)UInt32.MaxValue + 2, (SampleUInt32Enum)1)]
+        [InlineData((ulong)UInt32.MaxValue + 13, (SampleUInt32Enum)12)]
+        [InlineData((ulong)UInt32.MaxValue * 2, (SampleUInt32Enum)UInt32.MaxValue - 1)]
+        public static void ToString_UInt32(ulong i, SampleUInt32Enum enumValue)
```

Comment:
What about int32, int64, and uint64?

Reply:
Looking at the rest of the tests, I just realized we have the other numeric types below. It would be cleaner if they all followed the same naming/inline data pattern as this test.